src/elements/ifc-host-link.ts

```diff
@@ -0,0 +1,54 @@
+import { Client, EnvData } from '../client';
+const PATH_ATTR = 'path';
+
+/**
+ * A custom element responsible for rendering an anchor element, turning a path
+ * within the host application to a top-level url for the host application.
+ * It will be registered as `ifc-host-link` by a call to [[registerCustomElementsClient]] and
+ * should not be created directly. Instead add a `<ifc-host-link>` element to your
+ * markup or use `document.createElement('ifc-host-link')` after calling
+ * [[registerCustomElementsClient]].
+ */
+
+export default class IfcHostLinkElement extends HTMLElement {
+  private _client: Client;
+  private _environmentData: EnvData;
+
+  /** @internal */
+  constructor() {
+    super();
+    this._client = new Client();
+    this._environmentData = {
```

Comment:
Same as with the `ifc-client-link` component above, I don't know how to get the `environmentData` to the component. Here I just hard coded some values.